### PR TITLE
merge index.md with csharp.md, getting-started.md

### LINKED ISSUE #2507: Top result for C# on Google is a page not in TOC
Search results for [C# on Google](https://www.google.com/search?q=c%23) bring this page as its first result: https://docs.microsoft.com/en-us/dotnet/csharp/csharp

That page is not linked to the TOC, so we should remove it from the docset and redirect it to the [C# Guide](https://docs.microsoft.com/en-us/dotnet/csharp/).

But I'm wondering if we should have an introductory sentence about C# on the C# Guide similar to what the top result page has or what the F# Guide page has. @BillWagner?

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #2507

We had migrated landing pages that were never included in the TOC, but
were linked from several other merged pages.

This commit merges the migrated pages with the index page in the same
directory.  The directories in this PR are /docs/csharp, and
/docs/csharp/getting-started.

This is part of the work for Issue #2930

docs/csharp/csharp.md has been removed. Its content was included in docs/csharp/index.md. All internal links have been replaced, and a redirection record added.

docs/csharp/getting-started/getting-started-with-csharp.md has also been removed. Its content was included in docs/csharp/getting-started/index.md. Also, all links updated, and a redirection record added.

/cc @mairaw

### CODE REVIEW

File: .openpublishing.redirection.json [hunk 1]
```diff
@@ -939,6 +939,14 @@
...
             "source_path": "docs/csharp/getting-started/consuming-library-with-visual-studio.md",
             "redirect_url": "/dotnet/core/tutorials/consuming-library-with-visual-studio"
+        },
+        {
+            "source_path": "docs/csharp/csharp.md",
+            "redirect_url": "/dotnet/csharp"
+        },
+        {
+            "source_path": "docs/csharp/getting-started/getting-started-with-csharp.md",
+            "redirect_url": "/dotnet/csharp/getting-started/index.md"
```
Comment (mairaw) [inline .openpublishing.redirection.json]: redirect url doesn't contain md extension or it will fail

File: docs/csharp/getting-started/index.md [hunk 2]
```diff
@@ -5,18 +5,37 @@ description: Find short, simple tutorials that let you quickly learn introductor
 keywords: C#, Get Started, Acquisition, Install
 author: rpetrusha
 ms.author: ronpet
-ms.date: 05/02/2017
+ms.date: 08/23/2017
 ms.topic: article
 ms.prod: .net
 ms.technology: devlang-csharp
 ms.devlang: csharp
+translation.priority.ht:
```
Comment (mairaw) [inline docs/csharp/getting-started/index.md]: don't add this, let's follow up with loc on this metadata

But you need to bring the helpviewer_keywords from the other topics. They are for the offline book index.

File: docs/csharp/index.md [hunk 3]
```diff
@@ -4,12 +4,37 @@ description: Learn how the C# Guide can help you gain extensive knowledge on C#,
...
 author: BillWagner
 
-ms.date: 08/03/2016
+ms.date: 08/23/2017
 ms.topic: article
 ms.prod: .net
 ms.technology: devlang-csharp
 ms.devlang: csharp
 ms.assetid: 52db8280-0e53-40cf-858b-e8eef3997dea
+dev_langs:
```
Comment (mairaw) [inline docs/csharp/index.md]: I'm actually removing these from topics. Same for ms.devlang. They'll be set at a global level, unless the topic needs something different than the generic values. But C# will be the value for the csharp folder. The changes are sitting on a branch and I need to test them.

File: docs/csharp/index.md [hunk 4]
```diff
@@ -4,12 +4,37 @@ description: Learn how the C# Guide can help you gain extensive knowledge on C#,
...
 ms.devlang: csharp
 ms.assetid: 52db8280-0e53-40cf-858b-e8eef3997dea
+dev_langs: 
+  - "CSharp"
+helpviewer_keywords: 
+  - "documentation [C#]"
+  - "Help, Visual C# documentation"
+  - "C# language"
+  - "Visual C#"
+caps.latest.revision: 47
```
Comment (mairaw) [inline docs/csharp/index.md]: this is not from caps, so don't add this

File: docs/csharp/getting-started/index.md [hunk 5]
```diff
@@ -32,3 +51,33 @@ The following topics are available:
...
+* [Visual Studio Samples](/visualstudio/ide/visual-studio-samples)
+
+    Provides information about how you can access online samples.
+
+* [Walkthroughs](../../csharp/walkthroughs.md)
+
+    Provides links to programming walkthroughs that use C# and a brief description of each walkthrough.
+
+## See Also
+ [Getting Started with Visual C# and Visual Basic](/visualstudio/ide/getting-started-with-visual-csharp-and-visual-basic)
```
Comment (mairaw) [inline docs/csharp/getting-started/index.md]: this link caught me by surprise. we should probably do a PR on the VS repo to add VS to the title.

File: docs/csharp/index.md [hunk 6]
```diff
@@ -4,12 +4,37 @@ description: Learn how the C# Guide can help you gain extensive knowledge on C#,
...
+ms.author: "wiwagn"
+
+translation.priority.ht:
```
Comment (mairaw) [inline docs/csharp/index.md]: don't add these

**Review by mairaw — CHANGES_REQUESTED** [review]
I like this PR. Left some comments.

File: docs/csharp/getting-started/index.md (hunk 5 above)
Comment (BillWagner) [reply docs/csharp/getting-started/index.md]: Good idea.  See https://github.com/MicrosoftDocs/visualstudio-docs/pull/266

File: docs/csharp/index.md (hunk 4 above)
Comment (BillWagner) [reply docs/csharp/index.md]: I wasn't sure, because I did incorporate some of that content.

**Comment by BillWagner** [comment]
@mairaw I addressed all your comments. Please give it another look.

**Comment by BillWagner** [comment]
closing and reopening to restart the build.

**Review by mairaw — APPROVED** [review]
Changes look great! I just can't validate the build on staging. Is it working for you @BillWagner?

**Comment by BillWagner** [comment]
I can't see this PR on the review site. I'll force a new build.

**Comment by BillWagner** [comment]
build passed, the staging site is good, and I'll merge now.

**Pull request merged**